### PR TITLE
feat: Add the possibility to update the default handler for the Global Resource Property Registry.

### PR DESCRIPTION
**Pull request by armory-abedonik**
Unfortunately, it is not possible to override the default handler behavior from a Spinnaker plugin code. Therefore, we have decided to introduce a new setter function specifically designed to address this requirement.

### CODE REVIEW

**Review by sergio-quintero — APPROVED** [review]

**Comment by armory-abedonik** [comment]
Hi @dbyron-sf, please take a look at this pull request. Thank you in advance!

File: clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/description/GlobalResourcePropertyRegistry.java [hunk 1]
```diff
@@ -52,6 +52,11 @@ public GlobalResourcePropertyRegistry(
         new KubernetesResourceProperties(defaultHandler, defaultHandler.versioned());
   }
 
+  public void setDefaultHandler(KubernetesUnregisteredCustomResourceHandler defaultHandler) {
```
Comment (ovidiupopa07) [inline clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/description/GlobalResourcePropertyRegistry.java:55]: Would it make sense to check if the 'defaultHandler' is not null before creating the default properties?

**Review by ovidiupopa07 — CHANGES_REQUESTED** [review]

**Review by ovidiupopa07 — APPROVED** [review]

**Comment by sw-abedonik** [comment]
@mergifyio refresh

**Comment by ovidiupopa07** [comment]
@mergifyio refresh

**Review by dbyron-sf — APPROVED** [review]

**Comment by ovidiupopa07** [comment]
Because mergify doesn't have permission to update the repository, we manually merged the PR.

**Comment by sw-abedonik** [comment]
Hi @dbyron-sf, please push a new git tag based on the master branch for this repo. Thank you in advance!

**Comment by dbyron-sf** [comment]
Can you work with someone from Armory to take care of this please?

**Comment by edgarulg** [comment]
@Mergify backport release-1.31.x

**Pull request merged**